nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -416,7 +420,8 @@ in {
     config = {
       dbtype = mkOption {
         type = types.enum [ "sqlite" "pgsql" "mysql" ];
-        default = "sqlite";
+        default = if versionAtLeast "25.05" config.system.stateVersion then "pgsql" else "sqlite";
```

Comment:
Perhaps we should also put out an eval warning that sqlite is being used despite being unfit for production.

Though we can't really recommend existing users to do anything about it given that database migration is *still* broken upstream :/

The intention would be to prevent users like me who have an old stateVersion on their existing machine from unknowingly using sqlite when that's something you really don't want with (currently) not even a way out.

We could perhaps even leave the default value unset, requiring users to explicitly pick their poison.  
Existing users too would need to explicitly choose their current database which would theoretically be a "breaking" change but hardly a bad one and it's arguably better to be a little more explicit here anyways.

Reply:
> Perhaps we should also put out an eval warning that sqlite is being used despite being unfit for production.

Non-actioanble eval warnings are something I'd like to avoid, that only results in people ignoring warnings. There's this problem-infrastructure that we can probably use (I don't know if it's nixpkgs-only or also useful for nixos) if it gets merged at some point.

> The intention would be to prevent users like me who have an old stateVersion on their existing machine from unknowingly using sqlite when that's something you really don't want with (currently) not even a way out.

The release notes explicitly mention this. So I'd argue the next time people read these release-notes, they'll be aware (this is of course based on my _expectation_ that people read release-notes).

> Existing users too would need to explicitly choose their current database which would theoretically be a "breaking" change but hardly a bad

I mean that would only break people who haven't set their dbtype already, i.e. sqlite users only though. So perhaps that's the way to go.

Reply:
Yeah I'm most partial towards the last idea.